### PR TITLE
- Made it possible to load the config from the classpath

### PR DESCRIPTION
**Pull request by chrisdutz**
By using the URL class, we can not only use Files as config source, but also classpath resources.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -255,6 +255,11 @@
                 <artifactId>log4j</artifactId>
                 <version>1.2.17</version>
             </dependency>
+            <dependency>
```
Comment (jixuan1989) [inline pom.xml:258]: Why ... add LZ4 dependency here?

**Review by jixuan1989 — COMMENTED** [review]
The Test on WinOS failed... need to check why.
maybe because the current Integration Test mechanism can not delete the correct folders after this change.

File: pom.xml (hunk 1 above)
Comment (chrisdutz) [reply pom.xml:258]: I couldn't build IoTDB due to an error of the dependency plugin ... whithout this there were two different version versions in the reactor.

**Comment by chrisdutz** [comment]
That's strange ... I simply changed the way the InputStream is generated. If used with the previous config options, it should still have the same result.

**Comment by chrisdutz** [comment]
But while working on this ... I did notice the tests sometimes created directories and resources outside the "target" directory. This is considered not good practice as a failing test will prevent a cleanup ... in my case I couldn't do a new build till I found them all as otherwise the RAT test would have failed the build.

**Comment by jixuan1989** [comment]
> But while working on this ... I did notice the tests sometimes created directories and resources outside the "target" directory. This is considered not good practice as a failing test will prevent a cleanup ... in my case I couldn't do a new build till I found them all as otherwise the RAT test would have failed the build.

Agree with you. I will do the refactor in the following days.

File: pom.xml (hunk 1 above)
Comment (jixuan1989) [reply pom.xml:258]: OK, I checked the dependency tree, spark-core uses LZ4 v1.4  while others (e.g., tsfile) use v1.3

**Review by jixuan1989 — APPROVED** [review]

**Pull request merged**